### PR TITLE
fix: analyzer warnings SA1210, SA1414, VSTHRD002, CA1849, VSTHRD103, VSTHRD104

### PR DESCRIPTION
**Pull request by RushabhBhansali**
Related to #202 
- fix SA1210: Using directives should be ordered alphabetically

- fix SA1414: Tuple types in signatures should have element names

- fix VSTHRD002: synchronously waiting on Task

- fix CA1849: Call async methods when in an async method

- fix VSTHRD103: Call async methods when in an async method

### CODE REVIEW

File: src/Microsoft.ComponentDetection.Common/Telemetry/TelemetryRelay.cs [hunk 1]
```diff
@@ -60,10 +62,10 @@ public void Shutdown()
                 try
                 {
                     // Set a timeout for services that flush synchronously.
-                    AsyncExecution.ExecuteVoidWithTimeoutAsync(
+                    await AsyncExecution.ExecuteVoidWithTimeoutAsync(
                         () => service.Flush(),
```
Comment (JamieMagee) [inline src/Microsoft.ComponentDetection.Common/Telemetry/TelemetryRelay.cs:66]: Is there a `flushAsync()` available? And maybe the [`Task.Wait(TimeSpan)`](https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.wait?view=net-6.0#system-threading-tasks-task-wait(system-timespan)) that was added in .NET 6 is better?

Comment (JamieMagee) [reply src/Microsoft.ComponentDetection.Common/Telemetry/TelemetryRelay.cs:66]: We might need to create a followup issue to address `AsyncExecution.ExecuteVoidWithTimeoutAsync` calls.

File: src/Microsoft.ComponentDetection.Orchestrator/Orchestrator.cs [hunk 2]
```diff
@@ -202,8 +202,7 @@ private async Task GenerateEnvironmentSpecificTelemetryAsync(BcdeExecutionTeleme
 
                         return string.Join(Environment.NewLine, aptListResult.Split(Environment.NewLine).Where(x => x.Contains("libssl")));
                     });
-
-                    if (!getLibSslPackages.Wait(taskTimeout))
+                    if (await Task.WhenAny(getLibSslPackages, Task.Delay(taskTimeout)) != getLibSslPackages)
```
Comment (JamieMagee) [inline src/Microsoft.ComponentDetection.Orchestrator/Orchestrator.cs:205]: [`Task.WaitAsync(TimeSpan)`](https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.waitasync?view=net-7.0)?

**Review by JamieMagee — CHANGES_REQUESTED** [review]
Couple of comments on the async fixes

File: src/Microsoft.ComponentDetection.Common/Telemetry/TelemetryRelay.cs (hunk 1 above)
Comment (RushabhBhansali) [reply src/Microsoft.ComponentDetection.Common/Telemetry/TelemetryRelay.cs:66]: no `flushAsync()` is available here. 

the `AsyncExecution.ExecuteVoidWithTimeoutAsync()` has custom exceptions, may be not a good idea to change that logic here by using `Task.Wait(timespan)`.  We can probably keep the api of `ExecuteVoidWithtimeoutAsync()` same and use `Task.Wait(timespan)` inside of it (as part of the follow up issue) ?

File: src/Microsoft.ComponentDetection.Orchestrator/Orchestrator.cs (hunk 2 above)
Comment (RushabhBhansali) [reply src/Microsoft.ComponentDetection.Orchestrator/Orchestrator.cs]: done.

File: src/Microsoft.ComponentDetection.Orchestrator/Orchestrator.cs [hunk 3]
```diff
@@ -202,10 +202,10 @@ private async Task GenerateEnvironmentSpecificTelemetryAsync(BcdeExecutionTeleme
 
                         return string.Join(Environment.NewLine, aptListResult.Split(Environment.NewLine).Where(x => x.Contains("libssl")));
                     });
-
-                    if (!getLibSslPackages.Wait(taskTimeout))
+                    await getLibSslPackages.WaitAsync(taskTimeout);
+                    if (getLibSslPackages.IsFaulted)
                     {
-                        throw new TimeoutException($"The execution did not complete in the alotted time ({taskTimeout} seconds) and has been terminated prior to completion");
+                        throw new TimeoutException($"The execution did not complete in the alloted time ({taskTimeout} seconds) and has been terminated prior to completion");
```
Comment (JamieMagee) [inline src/Microsoft.ComponentDetection.Orchestrator/Orchestrator.cs:208]:
```suggestion
                        throw new TimeoutException($"The execution did not complete in the allotted time ({taskTimeout} seconds) and has been terminated prior to completion");
```

File: src/Microsoft.ComponentDetection.Orchestrator/Orchestrator.cs [hunk 4]
```diff
@@ -202,10 +202,10 @@ private async Task GenerateEnvironmentSpecificTelemetryAsync(BcdeExecutionTeleme
 
                         return string.Join(Environment.NewLine, aptListResult.Split(Environment.NewLine).Where(x => x.Contains("libssl")));
                     });
-
-                    if (!getLibSslPackages.Wait(taskTimeout))
+                    await getLibSslPackages.WaitAsync(taskTimeout);
+                    if (getLibSslPackages.IsFaulted)
```
Comment (JamieMagee) [inline src/Microsoft.ComponentDetection.Orchestrator/Orchestrator.cs:206]: I am not sure if this is correct. I think `WaitAsync` will throw `TaskCanceledException`.

**Review by JamieMagee — APPROVED** [review]

File: src/Microsoft.ComponentDetection.Common/Telemetry/TelemetryRelay.cs (hunk 1 above)
Comment (RushabhBhansali) [reply src/Microsoft.ComponentDetection.Common/Telemetry/TelemetryRelay.cs:66]: done

File: src/Microsoft.ComponentDetection.Orchestrator/Orchestrator.cs (hunk 3 above)
Comment (RushabhBhansali) [reply src/Microsoft.ComponentDetection.Orchestrator/Orchestrator.cs]: removed this  custom exception.

File: src/Microsoft.ComponentDetection.Orchestrator/Orchestrator.cs (hunk 4 above)
Comment (RushabhBhansali) [reply src/Microsoft.ComponentDetection.Orchestrator/Orchestrator.cs]: REmovved the duplicate exception.

**Pull request merged**